### ISSUE TITLE
Getting artifact:build error in artifact:build

### ISSUE THREAD

**Issue by shelane** [acquia/blt#4312]
**I want to...**
I have just updated from BLT 11.6.0 to 12.5.1 and all the code around it. I pushed my branch to my internal Bitbucket server and Jenkins sees a changes and tries to do a test build. We have a "Preparing Workspace" stage that executes the following:

```
                 sh "composer install --no-dev"
                 sh './vendor/bin/blt blt:telemetry:disable --no-interaction'
                 sh "./vendor/bin/blt artifact:build --no-interaction"
```

**It's not working because...**
I see that it is running the `artifact:build` command because this is in the log:
```
+ ./vendor/bin/blt artifact:build --no-interaction
Generating build artifact...
For more detailed output, use the -v flag.
```
[error]  There are no commands defined in the "artifact:build" namespace.
Did you mean one of these?
    artifact
    artifact:ac-hooks
    artifact:deploy
    artifact:install
    artifact:sync
    artifact:update
    artifact:update:drupal
    source:build 

I see it then going through the process of installing composer packages. At the end of that, I see this:
```
Scaffolding files for acquia/blt-project:
  - Skip [web-root]/sites/default/default.services.yml because it already exists and overwrite is false.
Updating BLT template files...
 > blt blt:update --ansi --no-interaction
Removing deprecated files and directories...
[33m> blt:init:shell-alias[39m
[32mThe BLT alias is already installed and up to date.[39m
PHP CodeSniffer Config installed_paths set to ../../acquia/coding-standards/src/Standards,../../drupal/coder/coder_sniffer,../../phpcompatibility/php-compatibility,../../sirbrillig/phpcs-variable-analysis,../../slevomat/coding-standard
[ExecStack] Done in 20.489s
Sanitizing artifact...
[error]  There are no commands defined in the "artifact:build" namespace.
```

Checking my logs for last successful build with BLT 11.6, I see this:
```
Running composer install --no-dev --no-interaction --optimize-autoloader in /opt/apps/jenkins/workspace/_drupal8_feature_DRUPAL8-184_css/deploy
```
but with 12.5.1, I see this:
```
Running composer install --no-dev --no-interaction --optimize-autoloader in /tmp/blt-deploy
```

So I started thinking that Jenkins had no idea what to do with /tmp/blt-deploy (and this would be especially bad if we had multiple branches running at the same time as it looks like it is not unique, but I digress).

Next, I tried changing the command to this:
```
./vendor/bin/blt artifact:build -D deploy.dir=deploy -D deploy.docroot=deploy/docroot --ansi --verbose --no-interaction
```
As it starts to attempt the build, I see this start appearing:
```
Sanitizing artifact...
[37;46;1m[info][39;49;22m Find Drupal core text files...
[37;46;1m[info][39;49;22m Find VCS directories...
[37;46;1m[info][39;49;22m Find .gitignore files...
[37;46;1m[info][39;49;22m Find Github directories...
[37;46;1m[info][39;49;22m Find INSTALL database text files...
[37;46;1m[info][39;49;22m Find other common text files...
[37;46;1m[info][39;49;22m Remove sanitized files from build...
[37;46;1m[Filesystem\FilesystemStack][39;49;22m [32mremove[39m [32m["/opt/apps/jenkins/workspace/rnal-drupal8_DRUPAL8-232-Drupal9/deploy/docroot/core/CHANGELOG.txt"][39m
[37;46;1m[Filesystem\FilesystemStack][39;49;22m [32mremove[39m [32m["/opt/apps/jenkins/workspace/rnal-drupal8_DRUPAL8-232-Drupal9/deploy/docroot/core/modules/migrate_drupal_ui/tests/src/Functional/d6/files/core/tests/fixtures/files/html-1.txt"][39m
[37;46;1m[Filesystem\FilesystemStack][39;49;22m [32mremove[39m [32m["/opt/apps/jenkins/workspace/rnal-drupal8_DRUPAL8-232-Drupal9/deploy/docroot/core/modules/migrate_drupal_ui/tests/src/Functional/d7/files/sites/default/files/ds9.txt"]
```
It never gets to a point of even beginning the composer install for the build. It just ends again with:
```
There are no commands defined in the "artifact:build" namespace.
```

**Detailed error output**
If a BLT command is failing, run it again with the `-vvv` argument and post the full output here.

**BLT doctor output**
BLT doctor likes to return critical errors that kill the Jenkins build, but at least it did output this data:
+---------------------------+--------------------------------------------------------------+
| Property                  | Value                                                        |
+---------------------------+--------------------------------------------------------------+
| blt-version               | 12.5.1.0                                                     |
| composer-version          | Composer version 2.0.3 2020-10-28 15:50:55                   |
| root                      | /opt/apps/jenkins/workspace/rnal-drupal8_DRUPAL8-232-Drupal9 |
|                           | /docroot                                                     |
| stacks.drupal-vm.inited   | false                                                        |
| stacks.dev-desktop.inited | false                                                        |
| uri                       | default                                                      |
+---------------------------+--------------------------------------------------------------+
+------------------------------------------+--------------------------------------------------------------+
| Check                                    | Problem                                                      |
+------------------------------------------+--------------------------------------------------------------+
| DbCheck:checkRequiredKeys                | drush status is missing the 'db-hostname' key.               |
| DrupalCheck:checkDrupalBootstrapped      | Could not bootstrap Drupal via drush without alias.          |
| DrupalCheck:checkDrupalInstalled         | Drupal is not installed.                                     |
|                                          |                                                              |
|                                          | Run `blt drupal:install` to install Drupal locally.          |
| FileSystemCheck:checkFileSystem:%files   | Public files directory is not set.                           |
| FileSystemCheck:checkFileSystem:%private | Private files directory is not set.                          |
| FileSystemCheck:checkFileSystem:%temp    | Temporary files directory is not set.                        |
| NodeCheck:checkNodeVersionFileExists     | Neither .nvmrc nor .node-version file found in repo root.    |
| PhpCheck:checkPhpDateTimezone            | PHP setting for date.timezone is not set.                    |
|                                          |   Define date.timezone in /etc/php.ini                       |
| SettingsFilesCheck:local-settings        | Could not find local settings file.                          |
|                                          | Your local settings file should exist at                     |
|                                          | /opt/apps/jenkins/workspace/rnal-drupal8_DRUPAL8-232-Drupal9 |
|                                          | /docroot/sites/default/settings/local.settings.php.          |
| SimpleSamlPhpCheck:checkSimpleSamlPhp    | Access to                                                    |
|                                          | /opt/apps/jenkins/workspace/rnal-drupal8_DRUPAL8-232-Drupal9 |
|                                          | /docroot/simplesaml is blocked by .htaccess                  |
|                                          |   Add the snippet in simplesamlphp-setup.md readme to your   |
|                                          | .htaccess file.                                              |
| WebUriCheck:checkUri                     | Site URI is not set                                          |
|                                          | Is options.uri set correctly in                              |
|                                          | /opt/apps/jenkins/workspace/rnal-drupal8_DRUPAL8-232-Drupal9 |
|                                          | /docroot/sites/default/local.drush.yml?                      |
+------------------------------------------+--------------------------------------------------------------+

**System information**
* Operating system type: Linux
* Operating system version: centos7
* BLT version: 12.5.1
* Jenkins running Docker with Pipeline steps

**Comment by shelane** [comment]
I have also tried adding this to my `blt.yml` file since that was what it was under BLT 11:

```
deploy:
  dir: ${repo.root}/deploy
```

There must be more to it that this because it is still failing.

Also, I have tried in my local to do a deploy:
```
blt artifact:deploy -n --commit-msg "Dev Deployment" --branch "d9-build" --no-interaction
```

But this results in the same error in the end. Here are the lines of output after the composer install is complete:
```
Updating BLT template files...
 > blt blt:update --ansi --no-interaction
Removing deprecated files and directories...
> blt:init:shell-alias
BLT can automatically create a Bash alias to make it easier to run BLT tasks.
This alias will be created in /home/docker/.bashrc.
Installing blt alias...
Added alias for blt to /home/docker/.bashrc.
You may now use the blt command from anywhere within a BLT-generated repository.
Restart your terminal session or run source /home/docker/.bashrc to use the new command.
PHP CodeSniffer Config installed_paths set to ../../acquia/coding-standards/src/Standards,../../drupal/coder/coder_sniffer,../../phpcompatibility/php-compatibility,../../sirbrillig/phpcs-variable-analysis,../../slevomat/coding-standard
[ExecStack] Done in 03:24
Sanitizing artifact...
[error]  There are no commands defined in the "artifact:build" namespace.

Did you mean one of these?
    artifact
    artifact:ac-hooks
    artifact:deploy
    artifact:install
    artifact:sync
    artifact:update
    artifact:update:drupal
    source:build 
```

**Comment by shelane** [comment]
I have found the issue:

The next command that the build command runs is `$this->deploySamlConfig();` and my `blt.yml` file has that setting to true, but I had missed the step of including the `blt-simplesamlphp` package. It is now building properly locally.

My Jenkins still isn't building properly, but that's a whole separate issue and different error that is coming up now. When trying to do an install it's reporting that I have PHP 7.2.9, when in fact, I know I have 7.3. I get this error:
```
In install.core.inc line 2301:
                                                                               
  [Drupal\Core\Installer\Exception\InstallerException]                         
  PHP: 7.2.9                                                                   
                                                                               
  Your PHP installation is too old. Drupal requires at least PHP <em class="p  
  laceholder">7.3.0</em>. It is recommended to upgrade to PHP version <em cla  
  ss="placeholder">7.3</em> or higher for the best ongoing support. See <a hr  
  ef="http://php.net/supported-versions.php">PHP's version support documentat  
  ion</a> and the <a href="https://www.drupal.org/docs/9/how-drupal-9-is-made  
  -and-what-is-included/environment-requirements-of-drupal-9#s-php-version-re  
  quirement">Environment requirements of Drupal 9</a> page for more informati  
  on.                                                                          
                                                                               
  Database settings:                                                           
                                                                               
  Array                                                                        
```

**Comment by b-sharpe** [comment]
Same thing, thanks for the direction. If anyone else is coming here from a D10 update which likely needed to switch away from simplesaml, you need to remove the following from blt.yml

```
simplesamlphp: true
```

**Comment by maxdmayhew** [comment]
this worked for me @b-sharpe. Thank you!

**Comment by shelane** [comment]
It looks like soon that we may not have to move away from simplesamlphp_auth: 
https://www.drupal.org/project/simplesamlphp_auth/issues/3349278
